datastores-parent/datastore-cassandra/src/main/java/org/wicketstuff/datastores/cassandra/CassandraDataStore.java

```diff
@@ -128,65 +149,72 @@ public CassandraDataStore(Cluster cluster, ICassandraSettings settings)
 		Select.Where dataSelect = QueryBuilder
-				.select(COLUMN_DATA)
-				.from(settings.getKeyspaceName(), settings.getTableName())
-				.where(QueryBuilder.eq(COLUMN_SESSION_ID, sessionId))
-				.and(QueryBuilder.eq(COLUMN_PAGE_ID, pageId));
+			.select(COLUMN_PAGE_TYPE, COLUMN_DATA)
+			.from(settings.getKeyspaceName(), settings.getTableName())
+			.where(QueryBuilder.eq(COLUMN_SESSION_ID, identifier))
+			.and(QueryBuilder.eq(COLUMN_PAGE_ID, pageId));
+		
 		ResultSet rows = session.execute(dataSelect);
 		Row row = rows.one();
-		byte[] bytes = null;
 		if (row != null)
 		{
+			String pageType = row.getString(COLUMN_PAGE_TYPE);
+			
 			ByteBuffer data = row.getBytes(COLUMN_DATA);
-			bytes = new byte[data.remaining()];
+			byte[] bytes = new byte[data.remaining()];
 			data.get(bytes);
+			
+			LOGGER.debug("Got data for session '{}' and page id '{}'", identifier, pageId);
 
-			LOGGER.debug("Got data for session '{}' and page id '{}'", sessionId, pageId);
+			return new SerializedPage(pageId, pageType, bytes);
 		}
-		return bytes;
+		
+		return null;
 	}
 
 	@Override
-	public void removeData(String sessionId, int pageId)
-	{
+	protected void removePersistedPage(String identifier, IManageablePage page) {
 		Delete.Where delete = QueryBuilder
 				.delete()
 				.all()
 				.from(settings.getKeyspaceName(), settings.getTableName())
-				.where(QueryBuilder.eq(COLUMN_SESSION_ID, sessionId))
-				.and(QueryBuilder.eq(COLUMN_PAGE_ID, pageId));
+				.where(QueryBuilder.eq(COLUMN_SESSION_ID, identifier))
+				.and(QueryBuilder.eq(COLUMN_PAGE_ID, page.getPageId()));
 		session.execute(delete);
 
-		LOGGER.debug("Deleted data for session '{}' and page with id '{}'", sessionId, pageId);
+		LOGGER.debug("Deleted data for session '{}' and page with id '{}'", identifier, page.getPageId());
 	}
 
 	@Override
-	public void removeData(String sessionId)
-	{
+	protected void removeAllPersistedPages(String identifier) {
 		Delete.Where delete = QueryBuilder
 				.delete()
 				.all()
 				.from(settings.getKeyspaceName(), settings.getTableName())
-				.where(QueryBuilder.eq(COLUMN_SESSION_ID, sessionId));
-		session.execute(delete);
+				.where(QueryBuilder.eq(COLUMN_SESSION_ID, identifier));
+			session.execute(delete);
 
-		LOGGER.debug("Deleted data for session '{}'", sessionId);
+			LOGGER.debug("Deleted data for session '{}'", identifier);
 	}
 
 	@Override
-	public void storeData(String sessionId, int pageId, byte[] data)
-	{
+	protected void addPersistedPage(String identifier, IManageablePage page) {
+		if (page instanceof SerializedPage == false)
+		{
+			throw new WicketRuntimeException("CassandraDataStore works with serialized pages only");
+		}
+		SerializedPage serializedPage = (SerializedPage)page;
+		
 		Insert insert = QueryBuilder
 				.insertInto(settings.getKeyspaceName(), settings.getTableName())
-				.using(QueryBuilder.ttl((int) settings.getRecordTtl().minutes()))
```

Comment:
In my tests I noticed that pages where expiring much too fast, so I adjusted the ttl to seconds, as it is specified on ttl().
Actually I didn't put to much thought into that change, since I expected that I screwed up that invocation by prior refactorings. But you might be right and this was a bug already.

Reply:
:+1: